pkg/process/config/config.go

```diff
@@ -311,17 +302,6 @@ func loadEnvVariables() {
 		}
 	}
 
-	// Support API_KEY and DD_API_KEY but prefer DD_API_KEY.
-	apiKey, envKey := os.Getenv("DD_API_KEY"), "DD_API_KEY"
-	if apiKey == "" {
-		apiKey, envKey = os.Getenv("API_KEY"), "API_KEY"
```

Comment:
Are we planning to deprecate the `API_KEY` env var? If I understood it correctly, with your updated it won't work anymore, right?

Reply:
Unfortouantely, this is correct that we will not support `API_KEY` anymore. However, I think the risk for this is low for a couple reasons:
1. The core agent doesn't support `API_KEY`, so if a customer used that environment variable, other agents would not work.
2. We have been warning them since 7.33 that `API_KEY` is an unknown environment variable.
3. `API_KEY` has been deprecated since this logic has been posted above.

Unfortunately would need to add special logic in the process agent only to have `API_KEY` override the api key, since it is unlikely that the core agent would want to support this.

WDYT?

Reply:
Thanks for clarifying it! The first point is indeed a good reason that allows us to deprecate the `API_KEY` env var. Should we just add it as a deprecation in the release note?

Reply:
Oh yeah, definitely. If we remove it it should be in the release notes! I will add a note for that